src/libs/logging/src/lib.rs

```diff
@@ -3,26 +3,46 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#[macro_use]
+extern crate lazy_static;
+use arc_swap::ArcSwap;
 use slog::{o, record_static, BorrowedKV, Drain, Key, OwnedKV, OwnedKVList, Record, KV};
 use std::collections::HashMap;
 use std::io;
 use std::io::Write;
 use std::process;
 use std::result;
-use std::sync::Mutex;
+use std::sync::{Arc};
 
 mod file_rotate;
 mod log_writer;
 
 pub use file_rotate::FileRotator;
 pub use log_writer::LogWriter;
 
+lazy_static! {
+    pub static ref FILTER_RULE: ArcSwap<HashMap<String, slog::Level>> =
+        ArcSwap::from(Arc::new(HashMap::from([
+            ("agent".to_string(), slog::Level::Info),
+            ("runtimes".to_string(), slog::Level::Info),
+            ("hypervisor".to_string(), slog::Level::Info),
+        ])));
+    pub static ref LOGGERS: ArcSwap<HashMap<String, slog::Logger>> =
+        ArcSwap::from(Arc::new(HashMap::new()));
+}
+
 #[macro_export]
 macro_rules! logger_with_subsystem {
     ($name: ident, $subsystem: expr) => {
         macro_rules! $name {
                             () => {
-                                    slog_scope::logger().new(slog::o!("subsystem" => $subsystem))
+                                match $subsystem {
+                                    "agent" => slog::Logger::clone(LOGGERS.load().get("agent").expect("Arccording logger should be found in map")),
+                                    "runtimes" => slog::Logger::clone(LOGGERS.load().get("runtimes").expect("Arccording logger should be found in map")),
+                                    "hypervisor" => slog::Logger::clone(LOGGERS.load().get("hypervisor").expect("Arccording logger should be found in map")),
+
+                                    _ => slog_scope::logger().new(slog::o!("subsystem" => $subsystem)),
+                                }
```

Comment:
The logging module is a universal module and should not know external subsystems. It is recommended to provide the function 'register_logger_for_subsystem' to add the subsystem by the register.